src/bun.js/webcore.zig

```diff
@@ -13,6 +14,279 @@ pub const Lifetime = enum {
+            // 2. Set message to the result of normalizing newlines given message.
+            // *  Not pertinent to a server runtime so we will just let the terminal handle this.
+
+            // 3. Set message to the result of optionally truncating message.
+            // *  Not necessary so we won't do it.
+            const message = arguments[0].?.value().toSlice(ctx.ptr(), bun.default_allocator);
+
+            stdout.writeAll(message.slice()) catch {
+                // 1. If we cannot show simple dialogs for this, then return null.
+                return JSC.JSValue.jsNull().asObjectRef();
+            };
+        }
+
+        // 4. Set default to the result of optionally truncating default.
+
+        // 5. Show message to the user, treating U+000A LF as a line break,
+        //    and ask the user to either respond with a string value or
+        //    abort. The response must be defaulted to the value given by
+        //    default.
+        stdout.writeAll(if (has_message) " " else "Prompt ") catch {
+            // 1. If we cannot show simple dialogs for this, then return false.
+            return JSC.JSValue.jsBoolean(false).asObjectRef();
+        };
+
+        // 6. Invoke WebDriver BiDi user prompt opened with this, "prompt" and message.
+        // *  Not relevant in a server context.
+
+        // 7. Pause while waiting for the user's response.
+        var stdin = std.io.getStdIn();
+        var reader = stdin.reader();
+
+        const first_byte = reader.readByte() catch {
+            // 8. Let result be null if the user aborts, or otherwise the string
+            //    that the user responded with.
+            return JSC.JSValue.jsNull().asObjectRef();
+        };
+
+        if (first_byte == '\n') {
+            // 8. Let result be null if the user aborts, or otherwise the string
+            //    that the user responded with.
+            return default;
+        }
+
+        var message = std.ArrayList(u8).initCapacity(bun.default_allocator, 1) catch {
+            // 8. Let result be null if the user aborts, or otherwise the string
+            //    that the user responded with.
+            return JSC.JSValue.jsNull().asObjectRef();
+        };
+
+        message.appendAssumeCapacity(first_byte);
+
+        readUntilDelimiterArrayListAppend(reader, &message, '\n', 1027) catch {
+            message.deinit();
+            // 8. Let result be null if the user aborts, or otherwise the string
+            //    that the user responded with.
+            return JSC.JSValue.jsNull().asObjectRef();
+        };
+
+        // 9. Invoke WebDriver BiDi user prompt closed with this, false if
+        //    result is null or true otherwise, and result.
+        // *  Too
+
+        // 8. If the user responded positively, return true; otherwise, the user
+        //    responded negatively: return false.
+        return JSC.ZigString.init(message.toOwnedSlice()).toValue(ctx.ptr()).asObjectRef();
```

Comment:
I'm fairly sure this is ok because the allocators should be able to share memory, right?

Reply:
This is confusing, but this memory will never be freed. `toValue` should really be named `toStaticValue`. Most of the places bun sends strings to JSC this way are statically allocated

The other issue is encoding. It has to be UTF16 or Latin1. 

Bun currently treats any non-ascii character as though it must be encoded to UTF-16. 

We can assume that these are going to be relatively small, so our various text encoding optimizations won't have much impact - we can let JSC handle the encoding here by calling `.markUTF8()` on the `ZigString` struct.

Also, to make our lives easier here, call `toValueGC(ctx).asObjectRef()` at the end and then bindings.cpp will clone the string into memory owned by JSC. 

Since we can do a stack allocation otherwise, then the only dynamic allocation majority of cases is the very last one